core/iwasm/aot/aot_loader.c

```diff
@@ -1148,12 +1182,207 @@ static void
 destroy_types(AOTType **types, uint32 count)
 {
     uint32 i;
-    for (i = 0; i < count; i++)
+    for (i = 0; i < count; i++) {
+
+#if WASM_ENABLE_GC != 0
+        if (types[i]->type_flag == WASM_TYPE_FUNC) {
+            AOTFuncType *func_type = (AOTFuncType *)types[i];
+            if (func_type->ref_type_map_count) {
```

Comment:
Had better use `if (func_type->ref_type_maps)`, since in the code below, `func_type->ref_type_map_count` is set before allocating memory for `func_type->ref_type_maps`, the `func_type->ref_type_map_count` may be larger than 0, but `func_type->ref_type_maps` is NULL (failed to allocate memory).